docs/reference/hacl/dh/ec/curve25519.md

```diff
@@ -0,0 +1,102 @@
+# Curve25519
+
+Elliptic-curve Diffie-Hellman key agreement on Curve25519.
+
+## Implementations
+
+There are multiple implementations, i.e., ...
+
+* a portable implementation (`Hacl_Curve25519_51.h`),
+* a fast x64 implementation that requires BMI2 & ADX CPU support (`Hacl_Curve25519_64.h`), and
+* a slower x64 implementation without the requirements stated above (`Hacl_Curve25519_64_Slow.h`).
```

Comment:
It's a 64-bit implementation but not specific to x64 systems, i.e. it can be compiled on x86 systems as well.